### PR TITLE
Use hoodie as plugin

### PR DESCRIPTION
**Pull request by gr2m**
adapt for changes introduced in https://github.com/hoodiehq/hoodie/releases

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
+var hapiOptions = {
+  server: {},
+  connection: {
+    port: 4201
+  }
+};
+var hoodieOptions = {
+  paths: {
+    public: 'dist'
+  }
```
Comment (gr2m) [inline index.js:22]: we will move these into Ember’s environment configuration later

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,6 @@
 /libpeerconnection.log
 npm-debug.log
 testem.log
+
+# hoodie
```
Comment (jameswestnz) [inline .gitignore:19]: Is this being saved for later by any chance?

File: index.js [hunk 3]
```diff
@@ -1,31 +1,48 @@
...
 function hoodieMiddleware(config) {
-  config.app.use('/hoodie', proxy({target: 'http://localhost:' + hoodie_options.port}));
+  config.app.use('/hoodie', proxy({target: 'http://localhost:' + hapiOptions.connection.port}));
+
+  var server = new Hapi.Server(hapiOptions.server);
+  server.connection(hapiOptions.connection);
 
-  getHoodieServer(hoodie_options, function (error, server, config) {
+  server.register({
+    register: hoodie,
```
Comment (jameswestnz) [inline index.js:32]: Should this be `hoodie.register`?

File: .gitignore (hunk 2 above)
Comment (gr2m) [inline .gitignore:19]: not sure what you mean? `# hoodie` is a comment to separate it from the sections above. What I added is ignoring the `.hoodie` folder, this is where the databases will be written into

File: index.js [hunk 4]
```diff
@@ -1,31 +1,48 @@
...
 
     console.log('app', 'Starting');
+    server.start(function (error) {
+      if (error) {
+        throw error;
+      }
 
-    server.start(function () {
-      console.log('Your Hoodie server has started on ' + url.format(config.connection));
+      console.log('Your Hoodie server has started on ' + url.format(server.info.uri));
```
Comment (jameswestnz) [inline index.js:45]: I'm not a (current) ember user myself, but if this is middleware, would you want to be logging this out each time? Is there somewhere else to report this?

File: .gitignore (hunk 2 above)
Comment (jameswestnz) [inline .gitignore:19]: My bad, I thought it was commenting out a mistake ;)

File: index.js (hunk 4 above)
Comment (gr2m) [inline index.js:45]: ¯\_(ツ)_/¯ good thinking, I don’t know, I guess we will find out eventually. I think it looks good for now, shouldn’t block this PR

Comment (jameswestnz) [inline index.js:45]: Totally, It's definitely not a blocker :)

File: index.js (hunk 3 above)
Comment (gr2m) [inline index.js:32]: good catch! Funny enough it seems to work like this, too. Looks like `hapi` has an internal check for these cases, cool!

Comment (jameswestnz) [inline index.js:32]: Oh sweet! Makes sense I suppose haha

**Comment by gr2m** [comment]
Thanks @jameswestnz

File: index.js (hunk 4 above)
Comment (jameswestnz) [inline index.js:45]: I don't use ember, but is this relevant? http://emberjs.com/api/classes/Ember.Logger.html

I was curious haha

**Pull request merged**